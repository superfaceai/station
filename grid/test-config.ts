import {
  NockConfig,
  SuperfaceTest,
  SuperfaceTestConfig,
} from '@superfaceai/testing';

export const nockConfig: NockConfig = {
  testInstance: expect,
};

/* eslint-disable-next-line @typescript-eslint/no-unsafe-member-access */
export const buildSuperfaceTest = (global as any).buildSuperfaceTest as (
  options?: SuperfaceTestConfig
) => SuperfaceTest;
