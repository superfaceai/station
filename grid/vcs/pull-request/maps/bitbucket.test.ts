import { SuperfaceTest } from '@superfaceai/testing';

import { nockConfig } from '../../../test-config';

describe(`vcs/pull-request/bitbucket}`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest(
      {
        profile: 'vcs/pull-request',
        provider: 'bitbucket',
      },
      nockConfig
    );
  });

  describe('PullRequest', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run({
          useCase: 'PullRequest',
          input: {
            owner: 'jakuvacek',
            repo: 'testrepository',
            identifier: 1,
          },
        })
      ).resolves.toMatchSnapshot();
    });

    it('should map error', async () => {
      await expect(
        superface.run({
          useCase: 'PullRequest',
          input: {
            owner: 'jakuvacek',
            repo: 'testrepository',
            identifier: 999,
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
