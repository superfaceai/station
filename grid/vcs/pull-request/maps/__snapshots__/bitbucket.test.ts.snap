// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcs/pull-request/bitbucket} PullRequest should map error 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"message\\": \\"Not Found\\",
  \\"statusCode\\": 404
}
AST Path: definitions[0].statements[3].responseHandlers[0].statements[1]
Original Map Location: Line 20, column 7",
}
`;

exports[`vcs/pull-request/bitbucket} PullRequest should perform successfully 1`] = `
Ok {
  "value": Object {
    "id": 1,
    "sha": "d1d6bab92584",
    "title": "README.md edited online with Bitbucket",
    "url": "https://bitbucket.org/jakuvacek/testrepository/pull-requests/1",
  },
}
`;
